app/controllers/va_facilities_controller.rb

```diff
@@ -1,30 +1,36 @@
 class VaFacilitiesController < ApplicationController
-  include PracticeUtils
+  include PracticeUtils, VaFacilitiesHelper
   before_action :set_va_facility, only: [:show, :created_practices, :update_practices_adopted_at_facility]
-
   def index
-    @facilities = VaFacility.cached_va_facilities.select(:common_name, :id, :official_station_name).order_by_station_name
+    @facilities = VaFacility.cached_va_facilities.select(:common_name, :id, :visn_id, :official_station_name).order_by_station_name.includes([:visn])
+    @clinical_resource_hubs = ClinicalResourceHub.all
+    @facilities = (@facilities.includes(:visn).sort_by(&:official_station_name.downcase) + @clinical_resource_hubs.includes([:visn]).sort_by(&:id))
     @visns = Visn.cached_visns.select(:name, :number)
     @types = VaFacility.cached_va_facilities.order_by_station_name.get_complexity
   end
 
   def load_facilities_index_rows
     if params[:facility].present?
       @facilities = [VaFacility.cached_va_facilities.order_by_station_name.includes([:visn]).find(params[:facility])]
+
+    elsif params[:crh].present?
+      @facilities = [ClinicalResourceHub.find_by_id(params[:crh].to_i)]
     else
       @facilities = VaFacility.cached_va_facilities.order_by_station_name.includes([:visn]).get_relevant_attributes
+      @clinical_resource_hubs = ClinicalResourceHub.all
 
       if params[:visn].present?
         @facilities = @facilities.where(visns: { number: params[:visn] })
+        visn_id = Visn.where(number: params["visn"].to_i).first().id
+        @clinical_resource_hubs = @clinical_resource_hubs.where(visn_id: visn_id).sort_by(&:id)
       end
-
       if params[:complexity].present?
+        @clinical_resource_hubs = []
```

Comment:
This is to clear all the clinical_resource_hubs... since they are not applicable when filtering by "complexity".  I am adding the the CRH array to the @facilities list on line 31..  so we only have to do this once for all the search filter scenarios.  fyi - I also added the cached_clinical_resource_hubs scope on line 20.